File: mobile/android/app/src/main/java/com/volttracker/obdpoc/BluetoothAdapters.java
package com.volttracker.obdpoc;

import android.bluetooth.BluetoothAdapter;
import android.bluetooth.BluetoothManager;
import android.content.Context;

final class BluetoothAdapters {

    private BluetoothAdapters() {}

    static BluetoothAdapter get(Context context) {
        if (context == null) {
            return null;
        }
        BluetoothManager manager = context.getSystemService(BluetoothManager.class);
        return manager == null ? null : manager.getAdapter();
    }
}

File: mobile/android/app/src/main/kotlin/com/volttracker/obdpoc/BluetoothAdapters.kt
package com.volttracker.obdpoc

import android.bluetooth.BluetoothAdapter
import android.bluetooth.BluetoothManager
import android.content.Context

object BluetoothAdapters {
    @JvmStatic
    fun get(context: Context?): BluetoothAdapter? {
        if (context == null) {
            return null
        }
        val manager = context.getSystemService(BluetoothManager::class.java)
        return manager?.adapter
    }
}
